pyccel/parser/semantic.py

```diff
@@ -853,8 +859,46 @@ def _visit_CommentBlock(self, expr, **settings):
         return expr
     def _visit_AnnotatedComment(self, expr, **settings):
         return expr
+
     def _visit_OmpAnnotatedComment(self, expr, **settings):
+        code = expr._user_nodes
+        code = code[-1]
+        index = code.body.index(expr)
+        combined_loop = expr.combined and ('for' in expr.combined or 'distribute' in expr.combined or 'taskloop' in expr.combined)
+
+        if isinstance(expr, (OMP_Sections_Construct, OMP_Single_Construct)) \
+           and expr.has_nowait:
+            for i, node in enumerate(code.body[index+1:], index + 1):
+                if isinstance(node, Omp_End_Clause):
+                    if node.txt.startswith(expr.name, 4):
+                        node.has_nowait = True
+
+        if isinstance(expr, (OMP_For_Loop, OMP_Simd_Construct,
+                            OMP_Distribute_Construct, OMP_TaskLoop_Construct)) or combined_loop:
+            msg = "Statement after {} must be a for loop.".format(type(expr).__name__)
+            if index == (len(code.body) - 1):
+                errors.report(msg, symbol=type(expr).__name__,
+                severity='fatal', blocker=self.blocking)
+
+            index += 1
+            while index < len(code.body):
```

Comment:
Some people consider it bad style to use `break`s. I think this is pretty clear but I wonder if the following would be clearer:
```
index += 1
while isinstance(code.body[index], (Comment, CommentBlock, Pass)) and index < len(code.body):
    index += 1

if index < len(code.body) and isinstance(code.body[index], For):
    if expr.has_nowait:
        nowait_expr = '!$omp end do'
        if expr.txt.startswith(' simd'):
            nowait_expr += ' simd'
        nowait_expr += ' nowait\n'
        code.body[index].nowait_expr = nowait_expr
else:
    errors.report(msg, symbol=type(code.body[index]).__name__,
        severity='fatal', blocker=self.blocking)
```
what do you think?

Reply:
yes, this is good